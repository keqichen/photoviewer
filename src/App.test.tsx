import React from 'react';
import App from './App';
import { ImageSelector } from './ImageSelector';
import { PhotoViewer } from './photoviewer/PhotoViewer';
import { render, screen, fireEvent } from '@testing-library/react';
import * as renderer from 'react-test-renderer';


test('renders React text', () => {
    const { getByText } = render(<App />);
    const textElement = getByText(/React/i);
    expect(textElement).toBeInTheDocument();
});

//A unit test to check our imageUrl generation code - 
// for me, this might check that the first link is 
// what I expect and that it doesn’t include the ‘broken’ images.

test('renders and checks PhotoViewer', () => {

    render(<PhotoViewer imgUrl={'https://picsum.photos/id/600/600/600.jpg'} />);
    const image = screen.getByRole("img")
    expect(image).toHaveAttribute('src', 'https://picsum.photos/id/600/600/600.jpg')
});

test('Check first image equal to what we are expecting', () => {
    render(<App />);
    const image = screen.getAllByRole('img')[0];
    expect(image).toHaveAttribute('src', 'https://picsum.photos/id/600/600/600.jpg')
});

//A ‘Regression / Snapshot' test 
//to confirm that I the ‘ImageViewer' component 
//doesn’t accidentally change in the future.

it('renders correctly', async () => {
    const tree = renderer.create(<PhotoViewer imgUrl={'https://picsum.photos/id/600/600/600.jpg'} />).toJSON();
    expect(tree).toMatchSnapshot();
});

it('renders correctly2', async () => {
    const tree = renderer.create(<App />).toJSON();
    expect(tree).toMatchSnapshot();
});

// A Component Test 
// to confirm that when I click a thumbnail
// then the thumbnail becomes selected 
// and the ImageViewer updates with the new image.

describe("My Component", () => {
    it("Should check the image has been selected", async () => {
        const component = render(<App />);

        const image1 = screen.getByAltText("this is an image")
        expect(image1).toHaveAttribute('src', 'https://picsum.photos/id/600/600/600.jpg')

        fireEvent.click(component.getByTestId("selected-element"));

        const image2 = screen.getByAltText("this is an image")
        expect(image2).toHaveAttribute('src', 'https://picsum.photos/id/602/600/600.jpg')
    });
});


