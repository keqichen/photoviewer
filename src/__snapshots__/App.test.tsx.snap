// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<div
  className="image-wrapper"
>
  <img
    alt="this is an image"
    className="image"
    src="https://picsum.photos/id/600/600/600.jpg"
  />
</div>
`;

exports[`renders correctly2 1`] = `
<div>
  <h1>
    React Photo Viewer
  </h1>
  <nav
    className="menu-wrapper"
  >
    <ul
      className="menu"
    >
      <li
        className="menu-items"
      >
        Home
      </li>
      <li
        className="menu-items"
      >
        Photos
      </li>
    </ul>
  </nav>
  <button
    className="button"
    onClick={[Function]}
  >
    Menu
  </button>
  <div
    className="image-wrapper"
  >
    <img
      alt="this is an image"
      className="image"
      src="https://picsum.photos/id/600/600/600.jpg"
    />
  </div>
  <h2>
    Select your photo
  </h2>
  <div
    className="selector-image"
  >
    <img
      className="thumbnail-image-selected thumbnail-image"
      data-testid=""
      onClick={[Function]}
      src="https://picsum.photos/id/600/600/600.jpg"
    />
    <img
      className="thumbnail-image"
      data-testid="selected-element"
      onClick={[Function]}
      src="https://picsum.photos/id/602/600/600.jpg"
    />
    <img
      className="thumbnail-image"
      data-testid=""
      onClick={[Function]}
      src="https://picsum.photos/id/603/600/600.jpg"
    />
    <img
      className="thumbnail-image"
      data-testid=""
      onClick={[Function]}
      src="https://picsum.photos/id/604/600/600.jpg"
    />
    <img
      className="thumbnail-image"
      data-testid=""
      onClick={[Function]}
      src="https://picsum.photos/id/605/600/600.jpg"
    />
    <img
      className="thumbnail-image"
      data-testid=""
      onClick={[Function]}
      src="https://picsum.photos/id/606/600/600.jpg"
    />
    <img
      className="thumbnail-image"
      data-testid=""
      onClick={[Function]}
      src="https://picsum.photos/id/607/600/600.jpg"
    />
    <img
      className="thumbnail-image"
      data-testid=""
      onClick={[Function]}
      src="https://picsum.photos/id/608/600/600.jpg"
    />
    <img
      className="thumbnail-image"
      data-testid=""
      onClick={[Function]}
      src="https://picsum.photos/id/609/600/600.jpg"
    />
    <img
      className="thumbnail-image"
      data-testid=""
      onClick={[Function]}
      src="https://picsum.photos/id/610/600/600.jpg"
    />
    <img
      className="thumbnail-image"
      data-testid=""
      onClick={[Function]}
      src="https://picsum.photos/id/611/600/600.jpg"
    />
    <img
      className="thumbnail-image"
      data-testid=""
      onClick={[Function]}
      src="https://picsum.photos/id/612/600/600.jpg"
    />
    <img
      className="thumbnail-image"
      data-testid=""
      onClick={[Function]}
      src="https://picsum.photos/id/613/600/600.jpg"
    />
    <img
      className="thumbnail-image"
      data-testid=""
      onClick={[Function]}
      src="https://picsum.photos/id/614/600/600.jpg"
    />
    <img
      className="thumbnail-image"
      data-testid=""
      onClick={[Function]}
      src="https://picsum.photos/id/615/600/600.jpg"
    />
    <img
      className="thumbnail-image"
      data-testid=""
      onClick={[Function]}
      src="https://picsum.photos/id/616/600/600.jpg"
    />
    <img
      className="thumbnail-image"
      data-testid=""
      onClick={[Function]}
      src="https://picsum.photos/id/617/600/600.jpg"
    />
    <img
      className="thumbnail-image"
      data-testid=""
      onClick={[Function]}
      src="https://picsum.photos/id/618/600/600.jpg"
    />
    <img
      className="thumbnail-image"
      data-testid=""
      onClick={[Function]}
      src="https://picsum.photos/id/619/600/600.jpg"
    />
    <img
      className="thumbnail-image"
      data-testid=""
      onClick={[Function]}
      src="https://picsum.photos/id/620/600/600.jpg"
    />
    <img
      className="thumbnail-image"
      data-testid=""
      onClick={[Function]}
      src="https://picsum.photos/id/621/600/600.jpg"
    />
    <img
      className="thumbnail-image"
      data-testid=""
      onClick={[Function]}
      src="https://picsum.photos/id/622/600/600.jpg"
    />
    <img
      className="thumbnail-image"
      data-testid=""
      onClick={[Function]}
      src="https://picsum.photos/id/623/600/600.jpg"
    />
    <img
      className="thumbnail-image"
      data-testid=""
      onClick={[Function]}
      src="https://picsum.photos/id/625/600/600.jpg"
    />
    <img
      className="thumbnail-image"
      data-testid=""
      onClick={[Function]}
      src="https://picsum.photos/id/626/600/600.jpg"
    />
    <img
      className="thumbnail-image"
      data-testid=""
      onClick={[Function]}
      src="https://picsum.photos/id/627/600/600.jpg"
    />
    <img
      className="thumbnail-image"
      data-testid=""
      onClick={[Function]}
      src="https://picsum.photos/id/628/600/600.jpg"
    />
    <img
      className="thumbnail-image"
      data-testid=""
      onClick={[Function]}
      src="https://picsum.photos/id/629/600/600.jpg"
    />
    <img
      className="thumbnail-image"
      data-testid=""
      onClick={[Function]}
      src="https://picsum.photos/id/630/600/600.jpg"
    />
    <img
      className="thumbnail-image"
      data-testid=""
      onClick={[Function]}
      src="https://picsum.photos/id/631/600/600.jpg"
    />
    <img
      className="thumbnail-image"
      data-testid=""
      onClick={[Function]}
      src="https://picsum.photos/id/633/600/600.jpg"
    />
    <img
      className="thumbnail-image"
      data-testid=""
      onClick={[Function]}
      src="https://picsum.photos/id/634/600/600.jpg"
    />
    <img
      className="thumbnail-image"
      data-testid=""
      onClick={[Function]}
      src="https://picsum.photos/id/635/600/600.jpg"
    />
    <img
      className="thumbnail-image"
      data-testid=""
      onClick={[Function]}
      src="https://picsum.photos/id/637/600/600.jpg"
    />
    <img
      className="thumbnail-image"
      data-testid=""
      onClick={[Function]}
      src="https://picsum.photos/id/638/600/600.jpg"
    />
    <img
      className="thumbnail-image"
      data-testid=""
      onClick={[Function]}
      src="https://picsum.photos/id/639/600/600.jpg"
    />
    <img
      className="thumbnail-image"
      data-testid=""
      onClick={[Function]}
      src="https://picsum.photos/id/640/600/600.jpg"
    />
    <img
      className="thumbnail-image"
      data-testid=""
      onClick={[Function]}
      src="https://picsum.photos/id/641/600/600.jpg"
    />
    <img
      className="thumbnail-image"
      data-testid=""
      onClick={[Function]}
      src="https://picsum.photos/id/642/600/600.jpg"
    />
    <img
      className="thumbnail-image"
      data-testid=""
      onClick={[Function]}
      src="https://picsum.photos/id/643/600/600.jpg"
    />
    <img
      className="thumbnail-image"
      data-testid=""
      onClick={[Function]}
      src="https://picsum.photos/id/645/600/600.jpg"
    />
    <img
      className="thumbnail-image"
      data-testid=""
      onClick={[Function]}
      src="https://picsum.photos/id/646/600/600.jpg"
    />
    <img
      className="thumbnail-image"
      data-testid=""
      onClick={[Function]}
      src="https://picsum.photos/id/648/600/600.jpg"
    />
    <img
      className="thumbnail-image"
      data-testid=""
      onClick={[Function]}
      src="https://picsum.photos/id/649/600/600.jpg"
    />
  </div>
</div>
`;
